test/parallel/test-vm-global-setter.js

```diff
@@ -0,0 +1,31 @@
+'use strict';
+require('../common');
+const assert = require('assert');
+const vm = require('vm');
+
+const window = createWindow();
+
+const descriptor =
+  Object.getOwnPropertyDescriptor(window.globalProxy, 'onhashchange');
+
+assert.strictEqual(typeof descriptor.get, 'function');
+assert.strictEqual(typeof descriptor.set, 'function');
+assert.strictEqual(descriptor.configurable, true);
+
+// Regression test for GH-42962. This assignment should not throw.
+window.globalProxy.onhashchange = () => {};
+
+function createWindow() {
+  const obj = {};
+  vm.createContext(obj);
+  Object.defineProperty(obj, 'onhashchange', {
+    // eslint-disable-next-line getter-return
+    get() {},
+    set() {},
```

Comment:
Should we also test if the setter is called in this case? (I suppose the expectation is that they do get called)

Reply:
Done!